mysticeti-core/src/syncer.rs

```diff
@@ -9,27 +9,48 @@ use crate::runtime::timestamp_utc;
 use crate::types::{AuthoritySet, RoundNumber, StatementBlock};
 use crate::{block_handler::BlockHandler, metrics::Metrics};
 use std::sync::Arc;
+use tokio::sync::watch::Sender;
 
-pub struct Syncer<H: BlockHandler, S: SyncerSignals, C: CommitObserver> {
+pub struct Syncer<H: BlockHandler, S: SyncerSignals, R: RoundAdvancedSignal, C: CommitObserver> {
     core: Core<H>,
     force_new_block: bool,
     commit_period: u64,
     signals: S,
     commit_observer: C,
     metrics: Arc<Metrics>,
+    round_advanced_signal: R,
 }
 
 pub trait SyncerSignals: Send + Sync {
     fn new_block_ready(&mut self);
 }
 
-impl<H: BlockHandler, S: SyncerSignals, C: CommitObserver> Syncer<H, S, C> {
+pub trait RoundAdvancedSignal: Send + Sync {
+    fn new_round(&mut self, round_number: RoundNumber);
+}
+
+impl RoundAdvancedSignal for Sender<RoundNumber> {
+    fn new_round(&mut self, round_number: RoundNumber) {
+        self.send(round_number).ok();
+    }
+}
+
+impl RoundAdvancedSignal for Option<RoundNumber> {
+    fn new_round(&mut self, round_number: RoundNumber) {
+        *self = Some(round_number)
+    }
+}
+
+impl<H: BlockHandler, S: SyncerSignals, R: RoundAdvancedSignal, C: CommitObserver>
```

Comment:
Do we actually need a new trait `RoundAdvancedSignal`? Could we instead add a method to `SyncerSignals` to keep this definition more concise? (we could introduce new struct to hold Notify and Sender that impl SyncerSignals)

Reply:
Ah I was really in between on that design but chose to break the signals. Let me though unify them, it seems that it makes more sense overall.